cmd/main.go

```diff
@@ -31,14 +31,17 @@ func init() {
 func main() {
 
 	options := Options{}
-	goflags := flags.NewParser(&options,
-		flags.Default)
+	goflags := flags.NewParser(&options, flags.Default)
 
 	parser := parser.Parser{}
 
 	_, err := goflags.Parse()
 	if err != nil {
-		logrus.Fatal("Error parsing flags")
+		if errFlags, ok := err.(*flags.Error); ok && errFlags.Type == flags.ErrHelp {
+			// flags automatically prints usage
```

Comment:
Does this still allow stdin input, in case no flags are provided?

Reply:
Yes. I'm specifically looking for `flags.ErrHelp`, because that means the help message was printed. It's [documented](https://pkg.go.dev/github.com/jessevdk/go-flags#Parser.ParseArgs) as follows:

> When the common help group has been added (AddHelp) and either -h or --help was specified in the command line arguments, a help message will be automatically printed if the PrintErrors option is enabled. Furthermore, the special error type ErrHelp is returned. It is up to the caller to exit the program if so desired.

There was a problem when reading from `stdin` that I fixed as followed (but did not yet commit):

```diff
diff --git a/cmd/main.go b/cmd/main.go
index 3a768a6..a14b612 100644
--- a/cmd/main.go
+++ b/cmd/main.go
@@ -49,15 +49,17 @@ func main() {
        }
 
        if options.Input == "" {
-               stat, err := os.Stdin.Stat()
+               _, err := os.Stdin.Stat()
 
                if err != nil {
                        logrus.Fatal("Error reading stdin")
                }
 
-               if ok := (stat.Size() > 0); !ok {
-                       logrus.Fatal("No input provided")
-               }
+               /*
+                       if ok := (stat.Size() > 0); !ok {
+                               logrus.Fatal("No input provided")
+                       }
+               */
 
                parser = *parser.NewParser(os.Stdin)
 
```

Reply:
In my tests, with the above diff applied, I ran `cat hetzner.json | ./cmd`, and that worked.

Reply:
neat!